spec/services/theme_settings_migrations_runner_spec.rb

```diff
@@ -0,0 +1,285 @@
+# frozen_string_literal: true
+
+describe ThemeSettingsMigrationsRunner do
+  fab!(:theme) { Fabricate(:theme) }
+  fab!(:migration_field) { Fabricate(:migration_theme_field, version: 1, theme: theme) }
+  fab!(:settings_field) { Fabricate(:settings_theme_field, theme: theme, value: <<~YAML) }
+      integer_setting: 1
+      boolean_setting: true
+      string_setting: ""
```

Comment:
I think we should test migrating list type as well as json schema types.

Reply:
There's a spec for the list type here: https://github.com/discourse/discourse/pull/24071/files#diff-2982249af72b3bf4f112ef04ff2a17049975e99480f2be551b5a958d85ce0b95R1081

JSON schema is not a type for some reasons -- it's technically a string type so there isn't really any difference from a migration's point of view.